File: sofd-core-opus-integration/src/main/java/dk/sofd/opus/service/model/Person.java
package dk.sofd.opus.service.model;

import java.time.LocalDate;
import java.util.HashSet;
import java.util.Set;
import java.util.stream.Collectors;

import com.fasterxml.jackson.annotation.JsonFormat;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

@Getter
@Setter
@Builder
@ToString                     // TODO: we need to implement some better API to support null'ing these
@EqualsAndHashCode(exclude = { "firstEmploymentDate", "anniversaryDate" })
@NoArgsConstructor
@AllArgsConstructor
public class Person {
	private String uuid;
	private String master;
	private boolean deleted;
	private String cpr;
	private String firstname;
	private String surname;
	
	@JsonFormat(pattern = "yyyy-MM-dd")
	private LocalDate firstEmploymentDate;
	
	@JsonFormat(pattern = "yyyy-MM-dd")
	private LocalDate anniversaryDate;
	
	private Post registeredPostAddress;
	private Post residencePostAddress;
	private Set<User> users;
	private Set<Affiliation> affiliations;
	
	public Set<Affiliation> onlyActiveAffiliations() {
		if (affiliations == null || affiliations.size() == 0) {
			return new HashSet<Affiliation>();
		}
		
		return affiliations.stream().filter(a ->
			(a.getStartDate() == null || a.getStartDate().compareTo(LocalDate.now().toString()) <= 0) &&
			(a.getStopDate() == null || a.getStopDate().compareTo(LocalDate.now().toString()) >= 0) &&
			 a.isDeleted() == false)
		.collect(Collectors.toSet());
	}
}

File: sofd-core-sd-integration/sofd-core-sd-integration/SOFD/Model/Person.cs
using DigitalIdentity.Utility;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;

namespace DigitalIdentity.SOFD.Model
{
    public class Person : ChangeTrackable
    {
        [JsonProperty("cpr")]
        public string Cpr { get; set; }

        [JsonProperty("master")]
        public string Master { get; set; }

        [JsonProperty("firstname")]
        public string Firstname { get; set; }

        [JsonProperty("surname")]
        public string Surname { get; set; }

        [JsonProperty("registeredPostAddress")]
        public PostAddress RegisteredPostAddress { get; set; }

        [JsonProperty("affiliations")]
        public List<Affiliation> Affiliations { get; set; } = new List<Affiliation>();

        [JsonProperty("firstEmploymentDate")]
        [JsonConverter(typeof(JsonDateConverter), "yyyy-MM-dd")]
        public DateTime? FirstEmploymentDate { get; set; }

        [JsonProperty("anniversaryDate")]
        [JsonConverter(typeof(JsonDateConverter), "yyyy-MM-dd")]
        public DateTime? AnniversaryDate { get; set; }

        [JsonProperty("uuid")]
        public string Uuid { get; set; }

        [JsonProperty("chosenName")]
        public string ChosenName { get; set; }

        public string GetName()
        {
            return ChosenName ?? $"{Firstname} {Surname}";
        }
        
    }
}
